SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -175,12 +242,39 @@ - (void)previewingContext:(id<UIViewControllerPreviewing>)previewingContext comm
     [self presentViewController:viewControllerToCommit animated:true completion:nil];
 }
 
+//MARK: - UISearchBarDelegate
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    [self.dataSource filterEventsWith:searchText];
+}
+
+- (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar {
+    [searchBar resignFirstResponder];
+    [self.dataSource filterEventsWith:searchBar.text];
+}
+
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
+    [searchBar resignFirstResponder];
+    [self.dataSource filterEventsWith:@""];
+    searchBar.text = @"";
+    [self.tableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0] atScrollPosition:UITableViewScrollPositionTop animated:YES];
+}
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
+    [searchBar resignFirstResponder];
+    [self.dataSource filterEventsWith:searchBar.text];
+}
+
 //MARK: - EventDataSourceDelegate
 
 - (void)willUpdateDataSource:(EventDataSource *)datasource {
     [self.tableView.refreshControl beginRefreshing];
 }
 
+- (void)reloadData {
```

Comment:
should this be `didUpdateDataSource:` to match `willUpdateDataSource:`? this way, the protocol could continue to say what happened, rather than asking for specific behavior